packages/hebao_v1/contracts/base/ERC1271Module.sol

```diff
@@ -42,35 +43,23 @@ abstract contract ERC1271Module is ERC1271, BaseModule
         methods[0] = this.isValidSignature.selector;
     }
 
-    function isSignedByWalletOwner(
-        address      _wallet,
+    // Will use msg.sender to detect the wallet, so this function should be called through
+    // the bounded method on the wallet itself, not directly on this module.
+    function isValidSignature(
         bytes memory _data,
-        bytes memory _signature
-        )
-        internal
+        bytes memory _signature)
+        public
         view
-        returns (bool)
-    {
-        return isSignedBy(Wallet(_wallet).owner(), _data, _signature);
-    }
-
-    function isSignedBy(
-        address      _address,
-        bytes memory _data,
-        bytes memory _signature
-        )
-        internal
-        pure
-        returns (bool)
+        override
+        returns (bytes4 magicValue)
     {
-        if (_address == address(0) || _signature.length != 65) {
-            return false;
+        bytes32 hash;
+        if(_data.length == 32) {
+            hash = BytesUtil.toBytes32(_data, 0);
+        } else {
+            hash = keccak256(_data);
         }
-
-        bytes32 signHash = keccak256(
-            abi.encodePacked("\x19Ethereum Signed Message:\n32", keccak256(_data))
-        );
-
-        return signHash.recoverECDSASigner(_signature) == _address;
+        hash.verifySignature(Wallet(msg.sender).owner(), _signature);
```

Comment:
should we use try-catch and make sure this method returns instead of throwing exception?

Reply:
Try/Catch can only be used for external calls unfortunately.

But I do agree that it would be cleaner if the function just returns true/false instead of throwing, but that needs quite some changes in `verifySignature`, but it's probably worth it do do that.

Reply:
Ok, lets add an TODO or issue then.

Reply:
Changed it so it returns a boolean. The function can still revert when the data is wrongly formatted, but I think that is acceptable.